src/main/java/com/minecolonies/network/messages/WorkOrderChangeMessage.java

```diff
@@ -0,0 +1,116 @@
+package com.minecolonies.network.messages;
+
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.colony.buildings.AbstractBuilding;
+import io.netty.buffer.ByteBuf;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler;
+import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
+
+/**
+ * Creates the WorkOrderChangeMessage which is responsible for changes in priority or removal of workOrders.
+ */
+public class WorkOrderChangeMessage implements IMessage, IMessageHandler<WorkOrderChangeMessage, IMessage>
+{
+    /**
+     * The Colony ID;
+     */
+    private int colonyId;
+
+    /**
+     * The workOrder to remove or change priority.
+     */
+    private int workOrderId;
+
+    /**
+     * The priority.
+     */
+    private int priority;
+
+    /**
+     * Remove the workOrder?
+     */
+    private boolean removeWorkOrder;
+
+    /**
+     * Empty public constructor
+     */
+    public WorkOrderChangeMessage()
+    {
+        /**
+         * Intentionally left empty.
+         **/
+    }
+
+    /**
+     * Creates object for the player to hire or fire a citizen.
+     *
+     * @param building  view of the building to read data from
+     * @param workOrderId     the workOrderId.
+     * @param removeWorkOrder remove the workOrder?
+     * @param priority the new priority.
+     */
+    public WorkOrderChangeMessage(AbstractBuilding.View building, int workOrderId, boolean removeWorkOrder, int priority)
+    {
+        this.colonyId = building.getColony().getID();
+        this.workOrderId = workOrderId;
+        this.removeWorkOrder = removeWorkOrder;
+        this.priority = priority;
+    }
+
+    /**
+     * Transformation to a byteStream.
+     *
+     * @param buf the used byteBuffer.
+     */
+    @Override
+    public void toBytes(ByteBuf buf)
+    {
+        buf.writeInt(colonyId);
+        buf.writeInt(workOrderId);
+        buf.writeInt(priority);
+        buf.writeBoolean(removeWorkOrder);
+    }
+
+    /**
+     * Transformation from a byteStream to the variables.
+     *
+     * @param buf the used byteBuffer.
+     */
+    @Override
+    public void fromBytes(ByteBuf buf)
+    {
+        colonyId = buf.readInt();
+        workOrderId = buf.readInt();
+        priority = buf.readInt();
+        removeWorkOrder = buf.readBoolean();
+    }
+
+    /**
+     * Called when a message has been received.
+     *
+     * @param message the message.
+     * @param ctx     the context.
+     * @return possible response, in this case -> null.
+     */
+    @Override
+    public IMessage onMessage(WorkOrderChangeMessage message, MessageContext ctx)
+    {
+        Colony colony = ColonyManager.getColony(message.colonyId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'colony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)